lua/entities/gmod_wire_expression2/core/selfaware.lua

```diff
@@ -106,12 +106,35 @@ e2function void setName( string name )
 	e:SetOverlayText(name)
 end
 
--- Get the name of another E2
+-- Set the name of a entity (component name if not E2). Thanks CaptainPRICE for idea and basic implementation
+e2function void entity:setName( string name )
+	if not IsValid(this) or E2Lib.getOwner(self, this) ~= self.player then return end
+	if this:GetClass() == "gmod_wire_expression2" then
+		if this.name == name then return end
+		if name == "generic" or name == "" then
+			name = "generic"
+			this.WireDebugName = "Expression 2"
+		else
+			this.WireDebugName = "E2 - " .. name
+		end
+		this.name = name
+		this:SetNWString( "name", this.name )
+		this:SetOverlayText(name)
+	else
+		if this.wireName == name or string.find(name, "[\n\r\"]") ~= nil then return end
```

Comment:
@AbigailBuccaneer
To your first comment: See the issue page where **I explained it why it has to be single-line**.
To your second comment: Yes, correct. This PR introduces a new variable `wireName` which is similar to `name` but it is for `WireName` networked-string instead. It is used to check for changes in order to prevent spamming. Btw, I have added anti-spam functionality yesterday, just haven't committed a new update, will do in a moment.